testv2/e2e/scenario_upgrade.go

```diff
@@ -45,24 +46,51 @@ func (scenario *scenarioUpgrade) SetVersions(versions ...string) {
 func (scenario *scenarioUpgrade) Run(t *testing.T) {
 	t.Helper()
 
-	install := scenarioInstall{
+	scenario.install = &scenarioInstall{
 		name:                 scenario.name,
 		manifestTemplatePath: scenario.manifestTemplatePath,
 		infra:                scenario.infra,
 		versions:             scenario.versions,
 	}
 
-	install.install(t)
+	scenario.install.install(t)
 	scenario.upgrade(t)
 	scenario.test(t)
 }
 
 func (scenario *scenarioUpgrade) upgrade(t *testing.T) {
-	// TODO: add upgrade logic
+	t.Helper()
+
+	k1 := newKubeoneBin(
+		scenario.infra.terraform.path,
+		renderManifest(t,
+			scenario.manifestTemplatePath,
+			manifestData{
+				VERSION: scenario.versions[1],
+			},
+		),
+	)
+
+	if err := k1.Apply(); err != nil {
+		t.Fatalf("kubeone apply failed: %v", err)
+	}
 }
 
 func (scenario *scenarioUpgrade) test(t *testing.T) {
-	// TODO: add some testings
+	t.Helper()
+
+	data := manifestData{
+		VERSION: scenario.versions[1],
```

Comment:
Should this eventually use the first version (i.e. `versions[0]`)?

Reply:
So if we take for example 1 upgrade test
```
func TestAwsDefaultsUpgradeContainerdFromV1_21_12_ToV1_22_9(t *testing.T) {
        infra := Infrastructures["aws_defaults"]
        scenario := Scenarios["upgrade_containerd"]
        scenario.SetInfra(infra)
        scenario.SetVersions("v1.21.12", "v1.22.9")
        scenario.Run(t)
}
```

we can se that versions[0] is OLD version, and versions[1] is NEW version. Obviously the versions[1] it used to verify the cluster after upgrade has been done.

Reply:
Oh, I see the `run` function now. Okay, it makes sense now.